contracts/PermissionedCauldron.sol

```diff
@@ -0,0 +1,96 @@
+// SPDX-License-Identifier: UNLICENSED
+
+pragma solidity 0.6.12;
+pragma experimental ABIEncoderV2;
+import "@boringcrypto/boring-solidity/contracts/libraries/BoringMath.sol";
+import "@boringcrypto/boring-solidity/contracts/BoringOwnable.sol";
+import "@boringcrypto/boring-solidity/contracts/ERC20.sol";
+import "@boringcrypto/boring-solidity/contracts/interfaces/IMasterContract.sol";
+import "@boringcrypto/boring-solidity/contracts/libraries/BoringRebase.sol";
+import "@boringcrypto/boring-solidity/contracts/libraries/BoringERC20.sol";
+import "@sushiswap/bentobox-sdk/contracts/IBentoBoxV1.sol";
+import "./NereusStableCoin.sol";
+import "./interfaces/IOracle.sol";
+import "./interfaces/ISwapper.sol";
+
+import "./WhitelistManager.sol";
+import "./CauldronV2.sol";
+
+
+
+/// @title PermissionedCauldron
+/// @dev This contract allows contract calls to any contract (except BentoBox)
+/// from arbitrary callers thus, don't trust calls from this contract in any circumstances.
+contract PermissionedCauldron is CauldronV2 {
```

Comment:
Not implemented permission management for all write operations, like accrue, addCollateral, removeCollateral, etc.